components/brave_federated/data_stores/data_store_unittest.cc

```diff
@@ -4,117 +4,189 @@
+  TrainingData TrainingDataFromTestInfo();
 
-  void AddAll();
+  bool AddTrainingInstance(std::vector<mojom::CovariatePtr> covariates);
+  void InitializeDataStore();
 
+  content::BrowserTaskEnvironment task_environment_;
   base::ScopedTempDir temp_dir_;
-  TestDataStore* test_data_store_;
+  DataStore* data_store_ = nullptr;
 };
 
 void DataStoreTest::SetUp() {
   ASSERT_TRUE(temp_dir_.CreateUniqueTempDir());
   base::FilePath db_path(
       temp_dir_.GetPath().Append(FILE_PATH_LITERAL("test_data_store")));
-  test_data_store_ = new TestDataStore(db_path);
-  ASSERT_TRUE(test_data_store_->Init(0, "test_federated_task", 50, 30));
-  ClearDB();
+  DataStoreTask data_store_task({0, "test_federated_task",
+                                 /* max_number_of_records */ 50,
+                                 base::Days(30)});
+  data_store_ = new DataStore(std::move(data_store_task), db_path);
+  ASSERT_TRUE(data_store_->InitializeDatabase());
 }
 
-void DataStoreTest::TearDown() {
-  test_data_store_ = nullptr;
+int DataStoreTest::RecordCount() const {
+  sql::Statement statement(data_store_->database_.GetUniqueStatement(
+      "SELECT count(*) FROM test_federated_task"));
+  EXPECT_TRUE(statement.Step());
+  return statement.ColumnInt(0);
 }
 
-void DataStoreTest::ClearDB() {
-  EXPECT_TRUE(test_data_store_->DeleteLogs());
+int DataStoreTest::TrainingInstanceCount() const {
+  sql::Statement statement(data_store_->database_.GetUniqueStatement(
+      "SELECT count(DISTINCT training_instance_id) FROM test_federated_task"));
+  EXPECT_TRUE(statement.Step());
+  return statement.ColumnInt(0);
 }
 
-size_t DataStoreTest::CountRecords() const {
-  sql::Statement s(test_data_store_->db_.GetUniqueStatement(
-      "SELECT count(*) FROM test_federated_task"));
-  EXPECT_TRUE(s.Step());
-  return static_cast<size_t>(s.ColumnInt(0));
+TrainingData DataStoreTest::TrainingDataFromTestInfo() {
+  TrainingData training_data;
+
+  for (const auto& item : kTrainingData) {
+    int training_instance_id = item.training_instance_id;
+    mojom::CovariatePtr covariate = mojom::Covariate::New();
+    covariate->type = (mojom::CovariateType)item.feature_name;
+    covariate->data_type = (mojom::DataType)item.feature_type;
+    covariate->value = item.feature_value;
+
+    training_data[training_instance_id].push_back(std::move(covariate));
+  }
+
+  return training_data;
 }
 
-TestTaskLog DataStoreTest::TestTaskLogFromTestInfo(
-    const TestTaskLogTestInfo& info) {
-  return TestTaskLog(info.id, info.label,
-                     base::Time::Now() - base::Days(info.days_from_now));
+void DataStoreTest::InitializeDataStore() {
+  TrainingData training_data = TrainingDataFromTestInfo();
+  for (auto& training_instance_pair : training_data) {
+    auto& training_instance = training_instance_pair.second;
+    AddTrainingInstance(std::move(training_instance));
+  }
+  EXPECT_EQ(std::size(kTrainingData), static_cast<size_t>(RecordCount()));
```

Comment:
std::size is c++17 and above. At this time we can only use up to C++14. See other usage in this PR. Thanks